gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.base;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.nessie.NessieCatalog;
+import org.apache.iceberg.spark.SparkSchemaUtil;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.types.StructType;
+import org.projectnessie.model.ImmutableTableReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+/** DDL + DML functionality for the "IdentifiedResult" table. */
+public final class IdentifiedResultsRepo {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IdentifiedResultsRepo.class);
+
+  private static final String COL_TYPE = "type";
+  private static final String COL_GC_RUN_START = "gcRunStart";
+  private static final String COL_GC_RUN_ID = "gcRunId";
+  private static final String COL_CONTENT_ID = "contentId";
+  private static final String COL_CONTENT_TYPE = "contentType";
+  private static final String COL_SNAPSHOT_ID = "snapshotId";
+  private static final String COL_METADATA_LOCATION = "metadataLocation";
```

Comment:
It's still wrong to even consider this value.
Nothing guarantees that the stored value will be valid wehn GC actually runs.

Reply:
oh. yeah global state changes.

Reply:
removed now.
 
Actually first time comment was just 
> Why is "metadataLocation" needed?

So, I replied why did I used it.  So, please provide a bit more context/expected result while working with me to avoid rework or re-review. Thanks.